src/components/PopupWithForm.js

```diff
@@ -22,11 +26,18 @@ export default class PopupWithForm extends Popup {
 
   setEventListeners() {
     super.setEventListeners();
-    this._popupForm.addEventListener("submit", this._handleSubmit);
+    this._popupForm.addEventListener("submit", (e) => {
+      e.preventDefault();
+      this.renderLoading();
```

Comment:
Needs correction: you should start the loading in index.js before the api request, since the form should not know anything about your submission logic (maybe you didn't want to use api)